tests/countries/test_antigua_and_barbuda.py

```diff
@@ -0,0 +1,186 @@
+        dt = (
+            "1995-01-02",
+            "2006-01-02",
+            "2012-01-02",
+            "2017-01-02",
+            "2023-01-02",
+        )
+        observed_name = f"{name} (observed)"
+        self.assertHolidayName(observed_name, dt)
+        self.assertNoNonObservedHoliday(dt)
+
+    def test_good_friday(self):
+        name = "Good Friday"
+        self.assertHolidayName(
+            name,
+            "2021-04-02",
+            "2022-04-15",
+            "2023-04-07",
+            "2024-03-29",
+            "2025-04-18",
+        )
+        self.assertHolidayName(name, range(1955, 2050))
+
+    def test_easter_monday(self):
+        name = "Easter Monday"
+        self.assertHolidayName(
+            name,
+            "2021-04-05",
+            "2022-04-18",
+            "2023-04-10",
+            "2024-04-01",
+            "2025-04-21",
+        )
+        self.assertHolidayName(name, range(1955, 2050))
+
+    def test_labour_day(self):
+        name = "Labour Day"
+        self.assertHolidayName(
+            name,
+            "2021-05-03",
+            "2022-05-02",
+            "2023-05-01",
+            "2024-05-06",
+            "2025-05-05",
+        )
+        self.assertHolidayName(name, range(1955, 2050))
+
+    def test_whit_monday(self):
+        name = "Whit Monday"
+        self.assertHolidayName(
+            name,
+            "2021-05-24",
+            "2022-06-06",
+            "2023-05-29",
+            "2024-05-20",
+            "2025-06-09",
+        )
+        self.assertHolidayName(name, range(1955, 2050))
+
+    def test_caribbean_community_day(self):
+        name = "Caribbean Community (Caricom) Day"
+        self.assertHolidayName(
+            name,
+            "1995-07-03",
+            "2000-07-03",
+            "2004-07-05",
+        )
+        self.assertNoHoliday(
+            "2005-07-04",
+            "2010-07-05",
+            "2015-07-06",
+            "2020-07-06",
+        )
+        self.assertHolidayName(name, range(1955, 2005))
+
+    def test_carnival_monday(self):
+        name = "Carnival Monday"
+        self.assertHolidayName(
+            name,
+            "2021-08-02",
+            "2022-08-01",
+            "2023-08-07",
+            "2024-08-05",
+            "2025-08-04",
+        )
+        self.assertHolidayName(name, range(1955, 2050))
+
+    def test_carnival_tuesday(self):
+        name = "Carnival Tuesday"
+        self.assertNoHoliday("2000-08-01", "2005-08-02")
+        self.assertHolidayName(
+            name,
+            "2021-08-03",
+            "2022-08-02",
+            "2023-08-08",
+            "2024-08-06",
+            "2025-08-05",
+        )
+        self.assertHolidayName(name, range(2006, 2050))
+
+    def test_independence_day(self):
+        name = "Independence Day"
+        self.assertHolidayName(name, (f"{year}-11-01" for year in range(1955, 2050)))
+        dt = ("2009-11-02", "2015-11-02", "2020-11-02")
```

Comment:
```suggestion
        dt = (
            "1992-11-02",
            "1998-11-02",
            "2008-11-03",
            "2009-11-02",
            "2014-11-03",
            "2015-11-02",
            "2020-11-02",
            "2025-11-03",
        )
```
More cases are needed to include years before and after 2005.